utils/ogmios-client/src/query_network.rs

```diff
@@ -0,0 +1,33 @@
+//! Queries that start with `queryNetwork/`.
+
+use crate::{types::SlotLength, OgmiosClient, OgmiosClientError, OgmiosParams};
+use serde::Deserialize;
+use serde_json::Value;
+use std::collections::HashMap;
+
+pub trait QueryNetwork: OgmiosClient {
+	#[allow(async_fn_in_trait)]
+	async fn shelley_genesis_configuration(
+		&self,
+	) -> Result<ShelleyGenesisConfigurationResponse, OgmiosClientError> {
+		let mut params = HashMap::new();
+		params.insert("era", Value::String("shelley".to_string()));
+		self.request("queryNetwork/genesisConfiguration", OgmiosParams::ByName(params))
+			.await
+	}
+}
+
+impl<T: OgmiosClient> QueryNetwork for T {}
+
+#[derive(Clone, Debug, Deserialize, PartialEq, Eq)]
+#[serde(rename_all = "camelCase")]
+pub struct ShelleyGenesisConfigurationResponse {
+	pub network_magic: u32,
+	pub security_parameter: u32,
+	// Ogmios returns active_slots_coefficient as string representing rational number, like "1/20"
+	pub active_slots_coefficient: String,
```

Comment:
Do we have to process/work with this coefficient? If not it's probably fine to leave it as String. Otherwise we might want to parse it into some adequate rational type?

Reply:
I used String and was going to leave it to user to chose his library, but I think I'll add `fraction` dependency and parse in the client.